src/Compilers/VisualBasic/Portable/Syntax/SyntaxNodeFactories.vb

```diff
@@ -196,10 +196,14 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
         ''' <param name="text">The input string</param>
         ''' <param name="offset">The starting offset in the string</param>
         Public Shared Function ParseTypeName(text As String, Optional offset As Integer = 0, Optional options As ParseOptions = Nothing, Optional consumeFullText As Boolean = True) As TypeSyntax
-            Using p = New InternalSyntax.Parser(MakeSourceText(text, offset), If(DirectCast(options, VisualBasicParseOptions), VisualBasicParseOptions.Default))
+            Dim vbOptions As VisualBasicParseOptions = DirectCast(options, VisualBasicParseOptions)
+            Using p = New InternalSyntax.Parser(MakeSourceText(text, offset), If(vbOptions, VisualBasicParseOptions.Default))
                 p.GetNextToken()
                 Dim node = p.ParseGeneralType()
-                Return DirectCast(If(consumeFullText, p.ConsumeUnexpectedTokens(node), node).CreateRed(Nothing, 0), TypeSyntax)
+                If consumeFullText Then
+                    node = p.ConsumeUnexpectedTokens(node)
+                End If
+                Return CreateRed(Of TypeSyntax)(node, p._scanner.Options)
```

Comment:
Are there more use sites that should be modified in a similar way? For example, ParseName on line 189.